azure/services/dns/arecords.go

```diff
@@ -151,13 +151,23 @@ func (s *Service) getDesiredARecords() []azure.ARecordSetSpec {
 			PublicIPName: s.scope.APIServerPrivateIP(),
 			TTL:          3600,
 		},
+			azure.ARecordSetSpec{
+				Hostname:     "apiserver",
+				PublicIPName: s.scope.APIServerPrivateIP(),
+				TTL:          3600,
+			},
 		)
 	case !s.scope.IsAPIServerPrivate():
 		aRecordSetSpec = append(aRecordSetSpec, azure.ARecordSetSpec{
 			Hostname:     "api",
 			PublicIPName: s.scope.APIServerPublicIP().Name,
 			TTL:          3600,
 		},
+			azure.ARecordSetSpec{
+				Hostname:     "apiserver",
+				PublicIPName: s.scope.APIServerPublicIP().Name,
+				TTL:          3600,
+			},
```

Comment:
```suggestion
```

we could also think about creating only one A record for the apiserver, depending on the mode:
* `api` for `public`
* `apiserver` for `private`

Reply:
I would go for both

Reply:
During the PR-creation this came into my mind and i wanted to share the option.
But i would also go for both as it doesn't hurt and we are independent for now.